app/controllers/web-payments/payment-auth-request.controller.js

```diff
@@ -43,11 +33,17 @@ module.exports = (req, res, next) => {
 
   return connectorClient({ correlationId: req.headers[CORRELATION_HEADER] }).chargeAuthWithWallet(chargeOptions, getLoggingFields(req))
     .then(data => {
-      setSessionVariable(req, `ch_${(chargeId)}.webPaymentAuthResponse`, {
-        statusCode: data.statusCode
-      })
+      if(data.body && data.body.error_identifier){
+        setSessionVariable(req, `ch_${(chargeId)}.webPaymentAuthResponse`, {
+          statusCode: data.statusCode, errorIdentifier: data.body.error_identifier
+        })
+      }else{
+        setSessionVariable(req, `ch_${(chargeId)}.webPaymentAuthResponse`, {
+          statusCode: data.statusCode
+        })
+      }
```

Comment:
A more concise way of doing this would be 

```suggestion
setSessionVariable(req, `ch_${(chargeId)}.webPaymentAuthResponse`, {
          statusCode: data.statusCode,
          ...data.body && data.body.error_identifier && { errorIdentifier: data.body.error_identifier }
        })
```